core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+            reviewer_id)]
+
+
+def get_suggestions_assigned_to_reviewer(assigned_reviewer_id):
+    """Gets a list of suggestions assigned to the given user for review.
+
+    Args:
+        assigned_reviewer_id: str. The ID of the reviewer assigned to review the
+                suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions assigned to the given user
+            for review.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel
+        .get_suggestions_assigned_to_reviewer(assigned_reviewer_id)]
+
+
+def get_suggestions_by_status(status):
+    """Gets a list of suggestions with the given status.
+
+    Args:
+        status: str. The status of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions with the given status.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_status(
+            status)]
+
+
+def get_suggestion_by_type(suggestion_type):
+    """Gets a list of suggestions with the given type.
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions of the given type.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_type(
+            suggestion_type)]
+
+
+def get_suggestions_by_target_id(target_type, target_id):
+    """Gets a list of suggestions to the entity with the given ID.
+
+    Args:
+        target_type: str. The type of target entity the suggestion is linked to.
+        target_id: str. The ID of the target entity the suggestion is linked to.
+
+    Returns:
+        list(Suggestion). A list of suggestions linked to the entity.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_target_id(
+            target_type, target_id)]
+
+
+def _update_suggestion(suggestion):
+    """Updates the given suggestion.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            updated.
+    """
+
+    suggestion_model = suggestion_models.SuggestionModel.get_by_id(
+        suggestion.suggestion_id)
+
+    suggestion_model.status = suggestion.status
+    suggestion_model.assigned_reviewer_id = suggestion.assigned_reviewer_id
+    suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
+    suggestion_model.change_cmd = suggestion.change_cmd
+    suggestion_model.score_category = suggestion.score_category
+
+    suggestion_model.put()
+
+
+def mark_review_completed(suggestion, status, reviewer_id):
+    """Marks that a review has been completed.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            updated.
+        status: str. The status of the suggestion post review.
```

Comment:
Perhaps list (and/or validate) the allowed choices

Reply:
Done